### ISSUE TITLE
Port is not always on 4370.

### ISSUE THREAD

**Issue by Zekt** [onetune/spotify-web-helper#6]
It's possible from 4370 to 4380 or more, current implementation will get nothing if Spotify's not open on port 4370(which happens all the time on my computer).

**Comment by JonnyBurger** [comment]
Interesting. Any idea how to find out what port SpotifyWebHelper runs on?
Should ports just be bruteforced until the correct port is found?

Are you interested in doing a PR?

**Comment by fejx** [comment]
Not 100% sure if this helps, but [this implementation](https://github.com/loretoparisi/node-spotify-webhelper) uses some kind of portscanner to determine the correct port, so I guess bruteforcing is the right choice. When I can I will try to look into it.

P.S. For me, it has worked ever since until Spotify updated yesterday, it seems the port has simply changed to 4371 now. My guess would be the first port that is checked by the webhelper has increased by 1 for some strange reason.

P.P.S. If you want to set another port manually, you can use `var helper = SpotifyWebHelper({port: 4371});` instead of changing the module.

**Comment by comfortme** [comment]
strangely stopped working for me too few days ago
i thought it might be related with the port but couldn't make it work.
i'm getting  `Error: read ECONNRESET at exports._errnoException (util.js:949:11) at TLSWrap.onread (net.js:563:26)`

**Comment by JonnyBurger** [comment]
This issue still hasn't affected me. 
I am accepting PRs for port-scanning though.

On macOS, I am able to find the port number using the following command:

`lsof -Pn -i4 | grep "SpotifyWe"`

![bildschirmfoto 2016-10-01 um 16 34 05](https://cloud.githubusercontent.com/assets/1629785/19014724/fbf37e6e-87f4-11e6-913f-b91dc6a2cfae.png)

Writing a parser that runs this command using `child_process` could potentially be more sophisticated than a port scanner.

**Comment by roncli** [comment]
Mentioned this in #7 ... the ports that it can use are 4370 to 4379 for HTTPS and 4380 to 4389 for HTTP.  This information comes from a script on the Spotify website.  This library only uses HTTPS, so you'd only need to check 4370 to 4379.

**Comment by Zekt** [comment]
It's not hard to reproduce though, just open something like `nc` on 4370 and 4371 port and see how Spotify web server behaves.

**Comment by TimStallard** [comment]
I've been playing around with this on Windows, and it seems that both 4370 and 4371 are used simultaneously (assuming they're available). SpotifyWebHelper seems to listen on 4370, and simply will not run if the port is taken. The Spotify application itself listens on 4371 and above - this is only listening if Spotify is running, but is available regardless of whether "Allow Spotify to be opened from the web" is enabled in the settings. Both of these two web servers seem to work with spotify-web-helper.

I've been able to test this with the latest version on Windows and Linux, which worked fine - I don't know if it's been implemented on OSX, though? If it has, I think this may simplify the entire project significantly, since SpotifyWebHelper wouldn't have to be started, but obviously that may affect backwards compatibility.

I was also able to use @JonnyBurger's method for finding ports, but I had to use the spotify process rather than SpotifyWebHelper. I did some testing using the NPM modules node-netstat and getpid, and I was able to get the Spotify port reliably, so that seems like it could work well, and is a much nicer solution that port scanning.

Edit: I've been working on this on TimStallard/spotify-web-helper, on the branch without-webhelper. I've changed the default port to 4371, and removed all of the WebHelper code, and it appears to be working as expected on both Windows and Linux.

Edit 2: A friend also tested this on OSX (on 4371), and it worked there as well.
